### ISSUE TITLE
study UE4 actor AI to make an enemy in the demo arena

### ISSUE THREAD

**Issue by mysterymagination** [mysterymagination/ryddelmyst#17]
https://learn.unrealengine.com/course/2814930/module/5839191?moduletoken=UHxxnDLPW8RutIMd9ShsNhh1mNCkfmTXX2-r-LBdQ5Z5viD77ceacS6x3p-CFJ6q&LPId=0

**Comment by mysterymagination** [comment]
So the main takeaway from the course above is that we need to subclass AIController (which has a PathFollowingComponent and an ActionsComponent, TBD how those will be involved in the CPP but presumably one or both are involved in the AI Move To node discussed later) -> create a TrackPlayer custom event -> create a timer at BeginPlay that emits the TrackPlayer event every 1 second -> Hook up an AI Move To node (TBD what equivalent CPP would look like) off TrackPlayer that moves to an acceptance radius around the target Actor, the player.

Translated to CPP, that is:
~~todo~~: create a TrackPlayer function that does whatever the AI Move To node does
~~todo~~: create a tracking timer with 1 second period that loops forever

**Comment by mysterymagination** [comment]
~~todo~~: add navmesh to Megyle so that reaper treant can receive nav info and move

**Comment by mysterymagination** [comment]
~~todo~~: why tf is the treant AI not running?  I created a basic twinstickshooter project to see if following along with the tutorial instructions more closely would help and it did -- take a look at diff between our Monster class and the BaseCharacter we created there.  Looks like probably we're missing some boilerplate necessary to get the AIController subclass to be active.
 -> thousands of years later, it was because I had a BeginPlay() override in the base CPP MonsterAI class and I forgot to call Super::BeginPlay() smh.
